File: experiments/014-LiveTypingFiveRuns/cells/1-Evaluate+TestRunning-University_8-DesignHeuristicsInline_free/2022-1c-recuperatorio-parcial1/20260917-040442-22085/exercise/java/src/main/java/customerimporter/CustomerSystem.java
package customerimporter;

public abstract class CustomerSystem {

    // customers

    public abstract void add(Customer aCustomer);

    public abstract Customer customerWithIdentificationType(String anIdType, String anIdNumber);

    public abstract int numberOfCustomers();

    // transactions

    public abstract void beginTransaction();

    public abstract void commit();

    // system lifecycle

    public abstract void start();

    public abstract void stop();
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-211617-6449/analysis/acceptance-project/src/customerimporter/customer_system.py
from abc import ABC, abstractmethod


class CustomerSystem(ABC):

    # customers

    @abstractmethod
    def add(self, a_customer):
        pass

    @abstractmethod
    def customer_with_identification_type(self, an_id_type, an_id_number):
        pass

    @abstractmethod
    def number_of_customers(self):
        pass

    # transactions

    @abstractmethod
    def begin_transaction(self):
        pass

    @abstractmethod
    def commit(self):
        pass

    # system lifecycle

    @abstractmethod
    def start(self):
        pass

    @abstractmethod
    def stop(self):
        pass
